### ISSUE TITLE
MediaListPlayer: Don't know about media type null

### ISSUE THREAD

**Issue by der-lange-christian** [caprica/vlcj#388]
I get the following Warning, is it a problem of my code or a warning I can ignore?

```
Jul 30, 2015 12:00:52 PM uk.co.caprica.vlcj.player.DefaultMediaPlayer$NotifyEventListenersRunnable run
WARNING: Event listener uk.co.caprica.vlcj.player.DefaultMediaPlayer$ResetMediaHandler@3b9f80a0 threw an exception Don't know about media type null
```

with this minimal example of MediaListPlayer

``` java
import java.io.UnsupportedEncodingException;
import uk.co.caprica.vlcj.medialist.MediaList;
import uk.co.caprica.vlcj.player.MediaPlayerFactory;
import uk.co.caprica.vlcj.player.embedded.EmbeddedMediaPlayer;
import uk.co.caprica.vlcj.player.list.MediaListPlayer;
import uk.co.caprica.vlcj.player.list.MediaListPlayerMode;

public class MediaListMinimal {

    public static void main(String[] args) throws UnsupportedEncodingException, InterruptedException {
        System.setProperty("java.util.logging.config.file", "logging.properties");

        MediaPlayerFactory mediaPlayerFactory = new MediaPlayerFactory("--no-video-title-show");

        EmbeddedMediaPlayer mediaPlayer = mediaPlayerFactory.newEmbeddedMediaPlayer();

        MediaListPlayer mediaListPlayer = mediaPlayerFactory.newMediaListPlayer();

        // Important: associate the media player with the media list player
        mediaListPlayer.setMediaPlayer(mediaPlayer);

        MediaList mediaList = mediaPlayerFactory.newMediaList();
        mediaList.addMedia("test-data/shortTrack2sec_1.mp3");
        mediaList.addMedia("test-data/shortTrack2sec_2.mp3");
        mediaList.addMedia("test-data/shortTrack2sec_3.mp3");


        mediaListPlayer.setMediaList(mediaList);
        mediaListPlayer.setMode(MediaListPlayerMode.LOOP);

        mediaListPlayer.play();

        Thread.currentThread().join();
    }
}
```

System:
- INFO: vlcj: 3.8.0
- INFO: java: 1.8.0_31 Oracle Corporation
- INFO: os: Linux 3.16.0-28-generic amd64
- INFO: vlc: 2.2.0-pre2 Weatherwax, changeset 2.2.0-pre1-15-g5178b24
- INFO: libvlc: /usr/lib/libvlc.so.5.5.0

**Comment by caprica** [comment]
I run the TestMediaListPlayer example in the vlcj sources and see no such error/warning.

**Comment by der-lange-christian** [comment]
When I add this two lines to the example I get the error also there:
```java
EmbeddedMediaPlayer mediaPlayer = mediaPlayerFactory.newEmbeddedMediaPlayer();
mediaListPlayer.setMediaPlayer(mediaPlayer);
```

I need the mediaPlayer to have control over the length and the current millisecond in the song

**Comment by caprica** [comment]
DId you look at BasicEmbeddedMediaListPlayerComponentTest ?

**Comment by caprica** [comment]
You can get the media player *from* the list player to get the length/millisecond, although for millisecond (position/time) you should use events.

**Comment by der-lange-christian** [comment]
I found only this Events:
```java
public class MediaListPlayerEventAdapter implements MediaListPlayerEventListener {

    // === Events relating to the media player ==================================

    @Override
    public void played(MediaListPlayer mediaListPlayer) {
    }

    @Override
    public void nextItem(MediaListPlayer mediaListPlayer, libvlc_media_t item, String itemMrl) {
    }

    @Override
    public void stopped(MediaListPlayer mediaListPlayer) {
    }

    // === Events relating to the current media =================================

    @Override
    public void mediaMetaChanged(MediaListPlayer mediaListPlayer, int metaType) {
    }

    @Override
    public void mediaSubItemAdded(MediaListPlayer mediaListPlayer, libvlc_media_t subItem) {
    }

    @Override
    public void mediaDurationChanged(MediaListPlayer mediaListPlayer, long newDuration) {
    }

    @Override
    public void mediaParsedChanged(MediaListPlayer mediaListPlayer, int newStatus) {
    }

    @Override
    public void mediaFreed(MediaListPlayer mediaListPlayer) {
    }

    @Override
    public void mediaStateChanged(MediaListPlayer mediaListPlayer, int newState) {
    }
}
```
Have I overseen something? I can not see length/millisecond interface?

The only geters I see on the MediaListPlayer are:

```java
getMediaList();
getMediaListPlayerState()
````

The EmbeddedMediaListPlayerComponent is for video content, is it? I only have mp3s to play.

**Comment by caprica** [comment]
1. In your comment above you mentioned adding EmbeddedMediaPlayer, so that's confusing things for me I suppose!

2. If you use AudioMediaListPlayerComponent, you can do getMediaPlayer() then add your event listeners.

**Comment by der-lange-christian** [comment]
thank you for the tip it looks interesting.

But with this code I get again the warnings:
```java
import uk.co.caprica.vlcj.component.AudioMediaListPlayerComponent;
import uk.co.caprica.vlcj.player.list.MediaListPlayerMode;


public class EmbeddedMediaPlayerComponent {
    
    public static void main(String[] args) throws InterruptedException {
        String mrl = "test-data/shortTrack2sec_1.mp3";
        AudioMediaListPlayerComponent mediaListPlayerComponent = new AudioMediaListPlayerComponent();
        // One line of vlcj code to add the media to the play-list...
        mediaListPlayerComponent.getMediaList().addMedia(mrl);
        // Another line of vlcj code to play the media...
        mediaListPlayerComponent.getMediaListPlayer().play();
        mediaListPlayerComponent.getMediaListPlayer().setMode(MediaListPlayerMode.LOOP);
        
        Thread.currentThread().join();
    }
}
```

Do you get them also?
```
Jul 30, 2015 9:57:40 PM uk.co.caprica.vlcj.player.DefaultMediaPlayer$NotifyEventListenersRunnable run
WARNING: Event listener uk.co.caprica.vlcj.player.DefaultMediaPlayer$ResetMediaHandler@6e3435b threw an exception Don't know about media type null
```

**Comment by caprica** [comment]
I didn't see them, but I didn't spend a lot of time on it.

I'll leave this issue open and investigate more in the coming days.

It's possible there are small bugs, because most people don't use the media list player.

**Comment by der-lange-christian** [comment]
okay thank you so far. You are really a big help. Without this kind of support I had looked for another solution ;) Thank you very much for your work.

**Comment by caprica** [comment]
I still want to find a better solution for #386 (actually #351) too...

**Comment by caprica** [comment]
I think the exception message you reported was because of a change to the media player itself, without consideration of the impact on the media list player.

**Comment by caprica** [comment]
Let's hope this was finally resolved as part of #690.
